src/platform_impl/linux/drm/event_loop.rs

```diff
@@ -0,0 +1,1417 @@
+            .ok_or(crate::error::OsError::new(
+                line!(),
+                file!(),
+                crate::platform_impl::OsError::DrmMisc("No connected connectors"),
+            ))?;
+
+        let crtc = crtcinfo.get(0).ok_or(crate::error::OsError::new(
+            line!(),
+            file!(),
+            crate::platform_impl::OsError::DrmMisc("No crtcs found"),
+        ))?;
+
+        // Get the first (usually best) mode
+        let &mode = con
+            .modes()
+            .iter()
+            .find(|f| f.mode_type().contains(ModeTypeFlags::PREFERRED))
+            .or(con.modes().get(0))
+            .ok_or(crate::error::OsError::new(
+                line!(),
+                file!(),
+                crate::platform_impl::OsError::DrmMisc("No modes found on connector"),
+            ))?;
+
+        let planes = drm.plane_handles().or(Err(crate::error::OsError::new(
+            line!(),
+            file!(),
+            crate::platform_impl::OsError::DrmMisc("Could not list planes"),
+        )))?;
+
+        /*
+          let mut db = drm
+          .create_dumb_buffer((64, 64), drm::buffer::DrmFourcc::Xrgb8888, 32)
+          .or(Err(crate::error::OsError::new(
+          line!(),
+          file!(),
+          crate::platform_impl::OsError::DrmMisc("Could not create dumb buffer"),
+          )))?;
+
+          {
+          let mut map = drm
+          .map_dumb_buffer(&mut db)
+          .expect("Could not map dumbbuffer");
+          for b in map.as_mut() {
+        *b = 128;
+        }
+        }
+
+        let fb = drm
+        .add_framebuffer(&db, 24, 32)
+        .or(Err(crate::error::OsError::new(
+        line!(),
+        file!(),
+        crate::platform_impl::OsError::DrmMisc("Could not create FB"),
+        )))?;
+
+        let (better_planes, compatible_planes): (
+        Vec<drm::control::plane::Handle>,
+        Vec<drm::control::plane::Handle>,
+        ) = planes
+        .planes()
+        .iter()
+        .filter(|&&plane| {
+        drm.get_plane(plane)
+        .map(|plane_info| {
+        let compatible_crtcs = res.filter_crtcs(plane_info.possible_crtcs());
+        compatible_crtcs.contains(&crtc.handle())
+        })
+        .unwrap_or(false)
+        })
+        .partition(|&&plane| {
+        if let Ok(props) = drm.get_properties(plane) {
+        let (ids, vals) = props.as_props_and_values();
+        for (&id, &val) in ids.iter().zip(vals.iter()) {
+        if let Ok(info) = drm.get_property(id) {
+        if info.name().to_str().map(|x| x == "type").unwrap_or(false) {
+        return val == (drm::control::PlaneType::Cursor as u32).into();
+        }
+        }
+        }
+        }
+        false
+        });
+        let plane = *better_planes.get(0).unwrap_or(&compatible_planes[0]);
+        */
+        let (p_better_planes, p_compatible_planes): (
+            Vec<drm::control::plane::Handle>,
+            Vec<drm::control::plane::Handle>,
+        ) = planes
```

Comment:
Could move plane creation to some function.

In general, I think it'd make sense to make code in event loop creation more declarative, and documented wrt steps. So it'll be easier for others to grasp it.

Reply:
Fixed